plugins/ftdidmx/FtdiDmxDevice.cpp

```diff
@@ -36,22 +41,45 @@ FtdiDmxDevice::FtdiDmxDevice(AbstractPlugin *owner,
     : Device(owner, widget_info.Description()),
       m_widget_info(widget_info),
       m_frequency(frequency) {
-  m_widget.reset(
+  m_widget =
       new FtdiWidget(widget_info.Serial(),
                      widget_info.Name(),
-                     widget_info.Id()));
+                     widget_info.Id(),
+                     widget_info.Vid(),
+                     widget_info.Pid());
 }
 
 FtdiDmxDevice::~FtdiDmxDevice() {
-  if (m_widget->IsOpen())
-    m_widget->Close();
+  delete m_widget;
 }
 
 bool FtdiDmxDevice::StartHook() {
-  AddPort(new FtdiDmxOutputPort(this,
-                                m_widget.get(),
-                                m_widget_info.Id(),
-                                m_frequency));
+  unsigned int interfaceCount = m_widget->GetInterfaceCount();
+  unsigned int succesfullyAdded = 0;
+
+  OLA_INFO << "Widget " << m_widget->Name() << " has " << interfaceCount
+           << " interfaces.";
+
+  for (unsigned int i = 1; i <= interfaceCount; i++) {
+    FtdiInterface *port = new FtdiInterface(m_widget,
+                                            static_cast<ftdi_interface>(i));
+    if (port->SetupOutput()) {
+      AddPort(new FtdiDmxOutputPort(this, port, i, m_frequency));
+      succesfullyAdded += 1;
+    } else {
+      OLA_WARN << "Failed to add interface: " << i;
+      delete port;
+    }
+  }
+  if (succesfullyAdded > 0) {
+    OLA_INFO << "Succesfully added " << succesfullyAdded << "/"
+             << interfaceCount << " interfaces.";
+  } else {
+    // I have not been able to simulate this case so far...
```

Comment:
Just hash out the increment to the counter variable and it will behave as if it failed to add them!